frontend/taipy-gui/src/components/Taipy/AutoLoadingTable.tsx

```diff
@@ -600,11 +601,12 @@ const AutoLoadingTable = (props: TaipyTableProps) => {
                         <MuiTable sx={tableSx} aria-labelledby="tableTitle" size={size} stickyHeader={true}>
                             <TableHead>
                                 <TableRow ref={headerRow}>
-                                    {colsOrder.map((col) => (
+                                    {colsOrder.map((col, index) => (
                                         <TableCell
                                             key={`head${columns[col].dfid}`}
                                             sortDirection={orderBy === columns[col].dfid && order}
                                             sx={columns[col].width ? { width: columns[col].width } : {}}
+                                            className={generateHeaderClassName(columns[col].title || columns[col].dfid, index)}
```

Comment:
- `columns[col].dfid` is enough
- No index please (it would be the responsability of the developper to handle identical class names
- use `getSuffixedClassNames(className, generateHeaderClassName(columns[col].dfid)` to get the current prefix